feat(docs): start any example in stackblitz

mqshaikh8 opened the pull request:
Closes #<issue>

## ✅ Checklist

- [✅] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [✅] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [✅] I performed a functional test on my final commit

---

## Changelog

I added an example page that lets users pick the different options and see how it would work on stackblitz.

---

## Screenshots

_[Screenshots]_
<img width="1402" alt="Screen Shot 2023-09-25 at 8 17 19 AM" src="https://github.com/t3-oss/create-t3-app/assets/46161085/b0fa0482-3e8d-4408-874f-4c3d95559f96">

💯

c-ehrlich commented on www/src/components/docs/exampleOptionForm.astro:
```diff
@@ -0,0 +1,91 @@
+          id={code}
+          type="checkbox"
+          name={name}
+          checked
```
since drizzle + prisma isn't a valid combination, could you make it so only drizzle is checked? dont care if its a condition on the checked prop or the loop is unwrapped.

c-ehrlich replied:
could maybe even add validation to the form so prisma and drizzle cant be selected together, but doesn't have to be part of this pr

mqshaikh8 replied:
Hi,
Sure, makes sense. I will keep one in this pr and then maybe make another pr where I add the other one with the conditional logic

c-ehrlich commented on www/src/components/docs/exampleOptionForm.astro:
```diff
@@ -0,0 +1,91 @@
+                    "version": "1.0.0",
+                    "description": "My Next.js app",
+                    "scripts":{
+                        "dev": " echo 'y' | npx create-t3-app example-app --CI ${checkedInputs} && cd ./my-app  && rm ../package.json " }
```
it's currently scaffolding in `./example-app` and then tries to cd to `./my-app`. i think out of the two i prefer `./example-app`

mqshaikh8 replied:
Yeah, 😅 That was a typo on my part. I'll get that fixed

c-ehrlich commented on pnpm-lock.yaml:
```diff
@@ -1,5 +1,9 @@
 lockfileVersion: '6.0'
 
+settings:
```
what are all of these changes to the lockfile? is it just that pnpm changed how some stuff works in a recent version?

mqshaikh8 replied:
This happened when I installed the stackblitz sdk using pnpm

juliusmarminge replied:
Can you make sure you're using pnpm@8.5.1 (as specified in package.json, then run

```
git checkout next pnpm-lock.yaml
pnpm i
```

That should remove any unnecessary diffs

c-ehrlich reviewed (changes requested):
thanks for the pr. this is a really clever way of implementing the demos :)

mqshaikh8 commented:
Hi,
It was a straightforward fix so I just added it in here. I also added a callout just to make sure the prisma and drizzle check switching is understood.

juliusmarminge commented on www/src/pages/en/examples.mdx:
Some info on this page that they must enter some environment variables in .env when their project is running if they want it to fully work / callout "some features might not work unless you create an env file" etc? Not sure what the best approach is

mqshaikh8 commented:
@juliusmarminge  I decided to add another callout to cover that.  I also am not able to get rod of the pnpm changes. I downgraded to 8.5.

mqshaikh8 commented:
@c-ehrlich @juliusmarminge  Let me know if there are other changes I should do.

juliusmarminge reviewed (approved):
This is awesome. It's not starting up the example automatically for me though:

![CleanShot 2023-10-06 at 10 48 42@2x](https://github.com/t3-oss/create-t3-app/assets/51714798/7756f51d-cae9-48b3-95a4-94554e268454)

If you wanna look into that as a follow-up PR that would be great, but no pressure as this is perfectly fine!
